src/WindowsUtils/include/WindowsUtils/ProcessLauncher.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WINDOWS_UTILS_PROCESS_LAUNCHER_H_
+#define WINDOWS_UTILS_PROCESS_LAUNCHER_H_
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/synchronization/mutex.h>
+
+#include "WindowsUtils/BusyLoopLauncher.h"
+
+namespace orbit_windows_utils {
+
+// Class used to launch a process, optionally pausing it at the entry point and
+// maintaining information required to suspend and resume the process main thread.
+class ProcessLauncher {
+ public:
+  ErrorMessageOr<uint32_t> LaunchProcess(const std::filesystem::path& executable,
+                                         const std::filesystem::path& working_directory,
+                                         const std::string_view arguments,
+                                         bool pause_at_entry_point);
+
+  ErrorMessageOr<void> SuspendProcess(uint32_t process_id);
```

Comment:
The interface is a bit weird here: You can only suspend / resume processes started with `pause_at_entry_point==true`, right? And the Suspend / Resume need to be called in that sequence and can only be called once. Do you even need the `pause_at_entry_point==false` case here? 

I don't have a great idea on how to enforce the sequence of events (could be done by introducing classes for LaunchedProcess, SuspendedProcess, ResumedProcess but that seems to complicate things way more than it helps). 
So maybe just add a comments on the public members here and maybe similar in BusyLauncher where the same applies.

Reply:
The sequence of events is enforced in the BusyLoopLauncher, which checks for the proper state transitions.